src/vm/builtins/builtin_runner/keccak.zig

```diff
@@ -265,53 +265,42 @@ pub const KeccakBuiltinRunner = struct {
     /// Calculates the Keccak hash of the input message.
     ///
     /// This function computes the Keccak hash of the provided input message and returns
-    /// it as an `ArrayList(u8)`. The Keccak hash function involves multiple steps of data
+    /// it as an fixed array based on block type. The Keccak hash function involves multiple steps of data
     /// processing.
     ///
     /// # Arguments
     ///
-    /// - `allocator`: An allocator for managing memory.
     /// - `input_message`: A pointer to the input message as an array of bytes.
     ///
     /// # Returns
     ///
-    /// An `ArrayList(u8)` containing the Keccak hash.
-    fn keccakF(allocator: Allocator, input_message: *[]const u8) !ArrayList(u8) {
-        var result = ArrayList(u8).init(allocator);
-        var vec = ArrayList(u64).init(allocator);
-        defer vec.deinit();
+    /// An fixed array based on block type, containing the Keccak hash.
+    /// current block type hardcoded to u64.
+    fn keccakF(input_message: *[]const u8) ![KeccakPrimitives.PLEN * @sizeOf(u64)]u8 {
+        var result = [_]u8{0} ** (KeccakPrimitives.PLEN * @sizeOf(u64));
+
+        // 1600 bits = 200 bytes = 25 u64
+        var st: std.crypto.core.keccak.KeccakF(@bitSizeOf(u64) * KeccakPrimitives.PLEN) = .{
+            .st = undefined,
+        };
+
+        const input = input_message.*;
 
         var i: usize = 0;
-        while (i + @sizeOf(u64) <= input_message.len) {
-            try vec.append(std.mem.readInt(
+        while (i + @sizeOf(u64) <= input.len) {
+            st.st[i / @sizeOf(u64)] = std.mem.readInt(
                 u64,
-                @ptrCast(input_message.*[i .. i + @sizeOf(u64)]),
+                @ptrCast(input[i .. i + @sizeOf(u64)]),
                 .little,
-            ));
+            );
+
             i += @sizeOf(u64);
         }
 
-        try vec.appendNTimes(
-            0,
-            KeccakPrimitives.PLEN - vec.items.len,
-        );
-
-        try KeccakPrimitives.keccak_p(
-            @ptrCast(
-                vec.items.ptr,
-            ),
-            KeccakPrimitives.keccakF_ROUND_COUNT,
-        );
+        st.permuteR(KeccakPrimitives.keccakF_ROUND_COUNT);
 
-        for (
-            @as(
-                *[KeccakPrimitives.PLEN]u64,
-                @ptrCast(vec.items.ptr),
-            ),
-        ) |item| {
-            var buf: [8]u8 = undefined;
-            std.mem.writeInt(u64, buf[0..], item, .little);
-            try result.appendSlice(&buf);
+        for (st.st, 0..) |item, idx| {
+            std.mem.writeInt(u64, result[idx * @sizeOf(u64) .. (idx + 1) * @sizeOf(u64)][0..@sizeOf(u64)], item, .little);
```

Comment:
Here as there are lots of arguments inside the `writeInt`, maybe put a comma at the end for multi lines fmt.